core/trino-main/src/main/java/io/trino/operator/TableFinishOperator.java

```diff
@@ -110,13 +111,15 @@ public OperatorFactory duplicate()
 
     private State state = State.RUNNING;
     private long rowCount;
-    private Optional<ConnectorOutputMetadata> outputMetadata = Optional.empty();
+    private final AtomicReference<ConnectorOutputMetadata> outputMetadata = new AtomicReference<>();
```

Comment:
just make `AtomicReference<Optional<ConnectorOutputMetadata>>` and avoid transition via `null`. `outputMetadata` is updated once anyway

Reply:
Changed